lib/collection/src/config.rs

```diff
@@ -116,6 +116,56 @@ impl CollectionParams {
             PayloadStorageType::InMemory
         }
     }
+
+    pub fn check_compatible(&self, other: &CollectionParams) -> CollectionResult<()> {
+        let CollectionParams {
+            vectors,
+            shard_number: _, // Maybe be updated by resharding, assume local shards needs to be dropped
+            sharding_method, // Not changeable
+            replication_factor: _, // May be changed
+            write_consistency_factor: _, // May be changed
+            read_fan_out_factor: _, // May be changed
+            on_disk_payload: _, // May be changed
+            sparse_vectors,  // Parameters may be changes, but not the structure
+        } = other;
+
+        self.vectors.check_compatible(vectors)?;
+
+        let mut this_sparse_vectors: Vec<_> = if let Some(sparse_vectors) = &self.sparse_vectors {
+            sparse_vectors.keys().collect()
+        } else {
+            vec![]
+        };
+        this_sparse_vectors.sort();
+
+        let mut other_sparse_vectors: Vec<_> = if let Some(sparse_vectors) = sparse_vectors {
+            sparse_vectors.keys().collect()
+        } else {
+            vec![]
+        };
+        other_sparse_vectors.sort();
```

Comment:
A hash set may be more fitting. No sorting required.

Reply:
agree